core/domain/user_jobs_one_off_test.py

```diff
@@ -2073,3 +2073,91 @@ def test_job_removes_deleted_exp_from_edited_explorations(self):
+        user_jobs_one_off.VerifyProfilePictureOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_mapreduce_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_mapreduce_tasks()
+        stringified_output = (
+            user_jobs_one_off.VerifyProfilePictureOneOffJob.get_output(job_id))
+        eval_output = [ast.literal_eval(stringified_item) for
+                       stringified_item in stringified_output]
+        return eval_output
+
+    def setUp(self):
+        def empty(*_):
+            """Function that takes any number of arguments and does nothing."""
+            pass
+
+        # We don't want to signup the superadmin user.
+        with self.swap(test_utils.TestBase, 'signup_superadmin_user', empty):
+            super(VerifyProfilePictureOneOffJobTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        user_services.generate_initial_profile_picture(self.owner_id)
+
+    def test_correct_profile_picture(self):
+        output = self._run_one_off_job()
+        self.assertEqual(output, [['CORRECT PROFILE PICTURE', 1]])
+
+    def test_profile_picture_non_standart_dimensions(self):
+        user_services.update_profile_picture_data_url(
+            self.owner_id, self.PNG_IMAGE_WRONG_DIMENSIONS_BASE64)
+        output = self._run_one_off_job()
+        self.assertEqual(
+            output, [['PROFILE PICTURE NON-STANDART DIMENSIONS - 150,160', 1]])
+
+    def test_cannot_load_profile_picture(self):
+        user_services.update_profile_picture_data_url(
+            self.owner_id, self.PNG_IMAGE_BROKEN_BASE64)
+        output = self._run_one_off_job()
+        self.assertEqual(output, [['CANNOT LOAD PROFILE PICTURE', 1]])
+
+    def test_profile_picture_not_png(self):
+        user_services.update_profile_picture_data_url(
+            self.owner_id, self.JPG_IMAGE_BASE64)
+        output = self._run_one_off_job()
+        self.assertEqual(output, [['PROFILE PICTURE NOT PNG', 1]])
+
+    def test_wrong_profile_picture_data_url(self):
+        user_services.update_profile_picture_data_url(
+            self.owner_id, self.BROKEN_BASE64)
+        output = self._run_one_off_job()
+        self.assertEqual(output, [['WRONG PROFILE PICTURE DATA URL', 1]])
+
+    def test_messing_profile_picture(self):
+        user_services.update_profile_picture_data_url(self.owner_id, None)
+        output = self._run_one_off_job()
+        self.assertEqual(output, [['MISSING PROFILE PICTURE', 1]])
+
+    def test_not_registered(self):
+        user_settings_model = (
+            user_models.UserSettingsModel.get_by_id(self.owner_id))
+        user_settings_model.username = None
+        user_settings_model.put()
+        output = self._run_one_off_job()
+        self.assertEqual(output, [['NOT REGISTERED', 1]])
+
+    def test_deleted(self):
```

Comment:
Suggest also testing the boundary case of no models (although it shouldn't happen in prod, it's useful to verify that the job does the right thing in such a case) & multiple users with different & same results (to make sure aggregation and cross-user checks happen correctly).